### ISSUE TITLE
Login Required Message not bubbling up

### ISSUE THREAD

**Issue by VED-StuartMorris** [auth0/auth0-react#163]
Hey,

Using the latest version here. Within our React SPA, we have a page called 'profile' in which a user can change their password.

For this functionality, we have a form that submits to a secure backend on our side with their access token. We first fetch the user token using the oauth/token endpoint with the username and current password, once confirmed we update the user's password using the Management API. 

This functionality is working as expected, although once the password change is complete we have so strange behaviour. 

On the form we have a Cancel button, if they hit this once the password change is complete, they are taken back to the route page.
This uses:
```onClick={() => history.goBack()}```
Result: They are taken back to the route and forced to login again. This works as expected, I guess...

We also have some breadcrumbs on the page, if they hit the link to return to the route, we get an error in ```auth0-react```.
```
Unhandled Rejection (Error): Login required
new OAuthError
src/errors.tsx:8

(anonymous function)
src/utils.tsx:15
```

The same behaviour happens if the user navigates back using the browser back button. Presumably, because the page is not actually refreshed the protected route part is not coming into play. 

**Expected behaviour**: Once the password change is complete and they navigate away from the profile page, they should be required to log in again.

Just for completeness...
The user object still contains the user, the error object in useAuth0 is undefined, the isAuthenticated is true and isLoading is false.

The question is, should this error bubble up to the error object in useAuth0 so that it can be handled? Or should we call a function once the password change is complete to force the user to login again?

Thanks

**Comment by adamjmcgrath** [comment]
Hi @VED-StuartMorris - we don't update the `error` object for methods you call yourself (eg `getTokenSilently`) you should catch the `login_required` error and prompt the user to login again (see the error handling in this example https://github.com/auth0/auth0-react/blob/master/EXAMPLES.md#4-create-a-useapi-hook-for-accessing-protected-apis-with-an-access-token). or you can preempt this by logging the user out once the password change is complete (you can do this locally using `logout({ localOnly: true })`)

**Comment by frederikprijck** [comment]
Closing this at it looks like it was answered. If not, please continue the conversation and we can re-open if neededn.

**Comment by VED-StuartMorris** [comment]
Hey @adamjmcgrath.

Resurrecting this issue as it appears to continue to occur. 

What we do at the moment is log the user out once the password change has been complete, however, we have a slight delay so the user can read a message that they should log in again.
```
    await updatePassword(formData, token)
      .then(async () => {
        setIsSaving(false);
        setSuccess(true);
        await delay(3000);
        logout();
      })
```

If the user navigates back or clicks a link that takes them back before the logout has occurred, they see the 'Login Required' error. 

On the page they navigate back to we have a different internal API call that uses the ```getAccessTokenSilently``` call. 

```
  useEffect(() => {
    async function fetchOrderHistoryData() {
      try {
        const token = await getAccessTokenSilently({
          audience: 'myaccountapi'
        });
        dispatch(fetchOrderHistory(token));
      } catch (error) {
        Log.error('Error occurred fetching order history', error);
      }
    }

    fetchOrderHistoryData();
  }, [dispatch, getAccessTokenSilently, refreshCount]);
```

What I notice in the above is that the token is returned. There is no error that bubbles up in the catch above. 

What happens is the error is shown, then the logout is eventually called and they are directed to log back in. We are trying to resolve/catch the login required earlier and handle that.

**Comment by VED-StuartMorris** [comment]
@adamjmcgrath @frederikprijck I do not have permission to reopen issues.

**Comment by frederikprijck** [comment]
> they see the 'Login Required' error.

What do you mean with they see this error? Where is it showing and where is it originating from?

**Comment by VED-StuartMorris** [comment]
The error is shown in the console which in turn is picked up in Sentry error reporting.

Of course, locally the error is shown in React.

0.chunk.js:3928 Uncaught (in promise) Error: Login required
    at new OAuthError (0.chunk.js:3928)
    at 0.chunk.js:3953
    at 0.chunk.js:4239
    at step (0.chunk.js:217)
    at Object.throw (0.chunk.js:147)
    at rejected (0.chunk.js:109)

**Comment by VED-StuartMorris** [comment]
![Screenshot 2021-01-08 at 10 59 14](https://user-images.githubusercontent.com/43371207/104007921-9a489f80-51a0-11eb-9c14-fd8132f96c0e.png)

**Comment by frederikprijck** [comment]
So the error isn't thrown by the `getAccessTokenSilently` code you are showing. Would u be able to provide a reproduction so we can see what is going on? You can also try and reproduce it in our playground application: https://github.com/auth0/auth0-react/blob/master/static/index.html

**Comment by VED-StuartMorris** [comment]
How is this static supposed to be used? It references a script that does not exist there <script src="auth0-react.js"></script>

**Comment by frederikprijck** [comment]
You should be able to run it using `npm start`.

**Comment by VED-StuartMorris** [comment]
presumably, pulling down the actual auth0-react repo running ```npm install``` and then ```npm start```? Just confused around the link to the index.html file. 

Just for additional information, just calling logout seems to cause this error

**Comment by VED-StuartMorris** [comment]
I think it would be a little difficult to reproduce this in the bare bones here. I can certainly show you this happening.

Do you have access to the Auth0 Slack, we could discuss this further on?

**Comment by adamjmcgrath** [comment]
Hey @VED-StuartMorris - sure, which Slack channel and I'll request to join

**Comment by VED-StuartMorris** [comment]
For visibility, we discussed this on a call. 

The issue was we had a ```getAccessTokenSilently``` call within a return of a useEffect. This was used to update the information we had about a user, once an update to the user was complete.

We had a flow which logged the user out, so this was called when a user was logged out. As adam explains above these errors should be caught and are not returned in the error object from useAuth0.

Therefore we just had to catch the error and ignore it (in our case).
```getAccessTokenSilently({ ignoreCache: true }).catch(error => {
        if (error.error !== 'login_required') { // catching the login required error which we can safely ignore here
          throw error;
        }
      });```
